lib/bundler/ui/silent.rb

```diff
@@ -13,6 +17,7 @@ def confirm(message, newline = nil)
       end
 
       def warn(message, newline = nil)
+        @warnings |= [message]
```

Comment:
Sorry, I accidentally removed @indirect's comment. Original comment was saying that (recovering from memories):

> Seems like this means message will never be garbage collected.